presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -76,7 +81,8 @@
     private final Metadata metadata;
     private final AccessControl accessControl;
     private final String catalogName;
-    private final Optional<Iterator<QualifiedTablePrefix>> tablePrefixIterator;
+    private final Optional<Iterator<QualifiedTablePrefix>> schemaPrefixIterator;
```

Comment:
I'm wondering if we can simplify this a bit, and reduce information being stored in multiple places for clarity. We already have the `catalogName` here, which we use for enumerating schemas to generate QualifiedTablePrefix. The main purpose of `schemaPrefixIterator` is to store schemas, since the catalog is common. And it does not store tables anymore. But we are using  `QualifiedTablePrefix` which is capable of storing 3 fields. Also, `tableNameDomain` is stored separately.

Is it possible to make this representation more compact? eg.
- using a String catalogName, a set<String> schemas and a set<String> tables ? OR
- using a set of QualifiedTablePrefixes. If we have table name filters (tableNameDomain), the   QualifiedTablePrefix will contain the table. If `tablePrefix.hasTableName` is false, then it would mean that we have to get output for all the tables. 

Do you foresee any issues with doing this? This might help us simplify the methods like `buildPagesColumns` too.

Reply:
Good point here. I've changed the implementation to use the the first approach (using a String catalogName, a set schemas and a set tables). I think it greatly improves clarity.